### ISSUE TITLE
Helmrelease with "Reconciling" condition when "Ready" is false 

### ISSUE THREAD

**Issue by Brotchu** [fluxcd/helm-controller#1083]
Reading https://fluxcd.io/flux/components/helm/helmreleases/#conditions, my understanding is 

when helmrelease is being reconciled, it will have 
- `Ready` condition with `Unknown` status
- `Reconciling` condition with `True` status

and when the reconciliation has failed, 
- `Ready` condition with `False` status
- `Stalled` condition with `True` status
- `Reconciling` condition removed

Is this the right interpretation?

Observed issue (if the above is right):
- `Ready` condition with `False` is present on helmrelease, from a failed upgrade
- a `Released` condition with `False` status is present as well, with Reason `UpgradeFailed`
- `Reconciling` condition  still present on helmrelease status, with status True

since `Reconciling` is still present in conditions, using kstatus `Compute`, the helmrelease object is inferred as `InProgress`, while it has failed the upgrade.

to reproduce:
I caused an upgrade failure from helm, setting it to fail at uninstall and then create the same helmrelease. 
flux version : 2.2.3

- create a helm chart with predelete hook that exits with 1 for failure.
- create a helmrelease that uses this helmchart
- delete the helmrelease, causing helm uninstall, which does not complete
- create the same helmrelease again

**Comment by stefanprodan** [comment]
The docs are for HelmRelease v2 GA, so you should be using Flux 2.4. The stalled condition is only set when the controller determines it can't progress, a failing upgrade doesn't mean it's stalled, depending on how you configured the HR it can retry, rollback, etc in all these cases the Reconciling condition is set to `ProgressingWithRetry`.

**Comment by Brotchu** [comment]
okay. thanks. 

I tried switching to flux 2.4 and see there is similar behavior on delete where it looks like controller gets into exponential backoff and retry (judging from the timestamp on reconciling conditions), and adds to the count of failures in status. 

Reading through similar discussions, I found https://github.com/fluxcd/flux2/discussions/119 , 
specifically the part 
```
Currently we set the ready condition like so:
- Read=Unknown signals that a reconciliation is underway
- Ready=True signals that a reconciliation has finished successfully
- Ready=False signals that a reconciliation has finished with errors
```

Is this still the same?  
If Ready=False here indicates error, irrespective of controller being Stalled or not, it might be easier to switch to checking Ready condition instead.
